addons/refuel/functions/fnc_readFuelCounter.sqf

```diff
@@ -18,24 +18,11 @@
 
 params [["_unit", objNull, [objNull]], ["_source", objNull, [objNull]]];
 
-[
-    TIME_PROGRESSBAR(REFUEL_PROGRESS_DURATION),
-    [_unit, _source],
-    {
-        params ["_args"];
-        _args params [["_unit", objNull, [objNull]], ["_source", objNull, [objNull]]];
-
-        private _currentFuel = [_source] call FUNC(getFuel);
-        if (_currentFuel == REFUEL_INFINITE_FUEL) then {
-            private _fuelCounter = 0.01 * round (100 * (_source getVariable [QGVAR(fuelCounter), 0]));
-            [[LSTRING(Hint_FuelCounter), _fuelCounter], 1.5, _unit] call EFUNC(common,displayTextStructured);
-        } else {
-            private _fuelCounter = 0.01 * round (100 * ((_source getVariable [QGVAR(fuelCounter), _currentFuel]) - _currentFuel));
-            [[LSTRING(Hint_FuelCounter), _fuelCounter], 1.5, _unit] call EFUNC(common,displayTextStructured);
-        };
-    },
-    "",
-    localize LSTRING(CheckFuelCounterAction),
-    {true},
-    [INTERACT_EXCEPTIONS]
-] call EFUNC(common,progressBar);
+private _currentFuel = [_source] call FUNC(getFuel);
+if (_currentFuel == REFUEL_INFINITE_FUEL) then {
+    private _fuelCounter = 0.01 * round (100 * (_source getVariable [QGVAR(fuelCounter), 0]));
+    [[LSTRING(Hint_FuelCounter), _fuelCounter], 1.5, _unit] call EFUNC(common,displayTextStructured);
+} else {
+    private _fuelCounter = 0.01 * round (100 * ((_source getVariable [QGVAR(fuelCounter), _currentFuel]) - _currentFuel));
```

Comment:
This is the only different thing in the entire statement. I suggest writing it as:
```sqf
private _currentFuel = [_source] call FUNC(getFuel);
private _fuelCounter = if (_currentFuel == REFUEL_INFINITE_FUEL) then {
    _source getVariable [QGVAR(fuelCounter), 0]
} else {
    (_source getVariable [QGVAR(fuelCounter), _currentFuel]) - _currentFuel
};

private _fuelCounter = 0.01 * round (100 * _fuelCounter);
[[LSTRING(Hint_FuelCounter), _fuelCounter], 1.5, _unit] call EFUNC(common,displayTextStructured);
```